client/app/views/file_view.coffee

```diff
@@ -135,18 +111,32 @@ module.exports = class FileView extends BaseView
             numUploadChildren = @uploadQueue.getNumUploadingElementsByPath path
             @hasUploadingChildren = numUploadChildren > 0
 
-            @listenTo @uploadQueue, 'add remove reset', =>
-                hasItems = @uploadQueue.getNumUploadingElementsByPath(path) > 0
-                if hasItems and not @$('.spinholder').is ':visible'
-                    @showLoading()
-                else if not hasItems and @$('.spinholder').is ':visible'
-                    @hideLoading()
 
-            @listenTo @uploadQueue, 'upload-complete', =>
-                @hasUploadingChildren = false
```

Comment:
Please don't merge, this is wrong, this should not be delegated, or at least not this way.